Add -NoPrompt option to suppress unnecessary prompts.

RossPatterson opened the pull request:
I didn't want to sit at the console and press RETURN to keep the process moving when there weren't any decisions to make.  So I added a -NoPrompt option to suppress the prompts.

wrljet commented:
Ross,

I see that I had mistakenly had some tabs in the files, which your changes expanded to 4 spaces instead of 8.
So now some of the indentation is goofed up.
Do you want to fix that?

Bill

RossPatterson commented:
Sure.  Too much Python recently, I guess 😀

What's your preference - tabstops at 8, 16, 24, _etc._?

wrljet commented:
Well, my intention is to keep Tab characters out of the files, because of this sort of thing.
But I clearly had left some in there.  I edit this stuff on so many different systems where I test it, they sneak in.

Indentation 4 spaces I think looks good.

Bill

RossPatterson commented:
>Well, my intention is to keep Tab characters out of the files, because of this sort of thing.

Yeah.  I'm an old mainframer, so I'm pro-space, anti-tab :-)

>But I clearly had left some in there. I edit this stuff on so many different systems where I test it, they sneak in.

No problem.  Python's anger over mixed tabs and spaces has sort of trained me to prevent them, but not always :-)

> Indentation 4 spaces I think looks good.

Cool.  I'll fix it up and re-push the PR.

RossPatterson commented:
Fixed and re-pushed.  `git diff -b` confirms that the only changes in this iteration are whitespace:

```bash
C:\Ross\Source\Hercules\hercules-helper-windows>git status
On branch noprompt
Your branch is up to date with 'origin/noprompt'.

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
        modified:   hercules-buildall.ps1
        modified:   hercules-step2.cmd

no changes added to commit (use "git add" and/or "git commit -a")

C:\Ross\Source\Hercules\hercules-helper-windows>git diff -b

C:\Ross\Source\Hercules\hercules-helper-windows>
```

There were a few debugging statements in `FindVCVARS()` that were commented out in column 1, I left those alone.  Ditto some `vcvars` stuff in the VS install section.  

There was a lot later in the main `try` block that was indented one or two levels too deep, I cleaned those up to match the depth of the code above them.

RossPatterson commented:
For completeness' sake, here is a `git diff -b` showing what the full PR is _without_ showing the effect of the whitespace changes.

```diff
diff --git a/hercules-buildall.ps1 b/hercules-buildall.ps1
index 6b6e4f3..fcd0db7 100644
--- a/hercules-buildall.ps1
+++ b/hercules-buildall.ps1
@@ -66,7 +66,10 @@ Param (
     [Switch]$Firewall,
 
     [Parameter(Mandatory = $false)]
-        [Switch]$ColorText
+    [Switch]$ColorText,
+
+    [Parameter(Mandatory = $false)]
+    [Switch]$NoPrompt
 )
 
 ##############################################################################
@@ -378,7 +381,7 @@ try {
     Write-Output ""
 
     Write-Output "==> Begin ..."
-    $input = Read-Host -Prompt 'Press return to continue'
+    if (!$NoPrompt) { $input = Read-Host -Prompt 'Press return to continue' }
 
     ##############################################################################
     # Check for existing VS2017 and required workloads
@@ -557,21 +560,21 @@ try {
 
     if ($vs_2017_missing -And $VS2017.IsPresent -And !$SkipVS.IsPresent) {
         Write-Output "==> Create/update VS2017 installer (this will take some time)"
-        $input = Read-Host -Prompt 'Press return to continue'
+        if (!$NoPrompt) { $input = Read-Host -Prompt 'Press return to continue' }
 
         # Create an offline installer for Visual Studio 2017
         .\create-vs2017-offline.ps1
         Write-Output ""
 
         Write-Output "==> Run VS2017 installer (this will take some time)"
-        $input = Read-Host -Prompt 'Press return to continue'
+        if (!$NoPrompt) { $input = Read-Host -Prompt 'Press return to continue' }
         pushd .\vs2017offline\
         cmd /c .\vs_community.exe --passive --norestart --wait
         popd
         Write-Output ""
     } elseif ($vs_2019_missing -And $VS2019.IsPresent -And !$SkipVS.IsPresent) {
         Write-Output "==> Create/update VS2019 installer (this will take some time)"
-        $input = Read-Host -Prompt 'Press return to continue'
+        if (!$NoPrompt) { $input = Read-Host -Prompt 'Press return to continue' }
 
         # Create an offline installer for Visual Studio 2019
         .\create-vs2019-offline.ps1
@@ -581,17 +584,17 @@ try {
 
         pushd .\vs2019offline\
         Write-Output "==> Run VS2019 installer to update (this will take some time)"
-          $input = Read-Host -Prompt 'Press return to continue'
+        if (!$NoPrompt) { $input = Read-Host -Prompt 'Press return to continue' }
         cmd /c .\vs_community.exe update --passive --norestart --wait
 
         Write-Output "==> Run VS2019 installer to add missing workloads (this will take some time)"
-          $input = Read-Host -Prompt 'Press return to continue'
+        if (!$NoPrompt) { $input = Read-Host -Prompt 'Press return to continue' }
         cmd /c .\vs_community.exe --passive --norestart --wait
         popd
         Write-Output ""
     } elseif ($vs_2022_missing -And $VS2022.IsPresent -And !$SkipVS.IsPresent) {
         Write-Output "==> Create/update VS2022 installer (this will take some time)"
-        $input = Read-Host -Prompt 'Press return to continue'
+        if (!$NoPrompt) { $input = Read-Host -Prompt 'Press return to continue' }
 
         # Create an offline installer for Visual Studio 2022
         .\create-vs2022-offline.ps1
@@ -601,11 +604,11 @@ try {
 
         pushd .\vs2022offline\
         Write-Output "==> Run VS2022 installer to update (this will take some time)"
-          $input = Read-Host -Prompt 'Press return to continue'
+        if (!$NoPrompt) { $input = Read-Host -Prompt 'Press return to continue' }
         cmd /c .\vs_community.exe update --passive --norestart --wait
 
         Write-Output "==> Run VS2022 installer to add missing workloads (this will take some time)"
-          $input = Read-Host -Prompt 'Press return to continue'
+        if (!$NoPrompt) { $input = Read-Host -Prompt 'Press return to continue' }
         cmd /c .\vs_community.exe --passive --norestart --wait
         popd
         Write-Output ""
@@ -652,7 +655,7 @@ try {
     "set HERCULES_HELPER_BUILD_DIR=$hercules_dir" | Out-File -FilePath $rebuild_filename -Append
     "set HERCULES_HELPER_VCVARS_CMD=$vcvars_cmd" | Out-File -FilePath $rebuild_filename -Append
 
-    $input = Read-Host -Prompt 'Press return to continue'
+    if (!$NoPrompt) { $input = Read-Host -Prompt 'Press return to continue' }
 
     $props_dir = "$HOME\AppData\Local\Microsoft\MSBuild\v4.0"
     Write-Output "Visual Studio User Properties directory: $props_dir"
@@ -708,7 +711,7 @@ try {
     pushd "$hercules_dir"
 
     Write-Output "==> Clone Hercules from GitHub and download other packages"
-    $input = Read-Host -Prompt 'Press return to continue'
+    if (!$NoPrompt) { $input = Read-Host -Prompt 'Press return to continue' }
 
     # If hyperion repo directory already exists, and -ForceClone is
     # specified, delete it.
@@ -853,15 +856,15 @@ try {
     # cmd /k "$vcvars_cmd"
     # Invoke-Expression -Command "cmd /c hercules-step2.cmd"
     # cmd.exe /c hercules-step2.cmd 2`>`&1 | Tee-Object -FilePath "hercules-helper-build.log"
-    cmd.exe /c hercules-step2.cmd $Flavor 2`>`&1 | Tee-Object -Variable dummy
+    cmd.exe /c hercules-step2.cmd $Flavor $NoPrompt 2`>`&1 | Tee-Object -Variable dummy
 
     Write-Output "... back from build steps"
     Write-Output ""
 
     ##############################################################################
     if ($Firewall.IsPresent) {
-        Write-Output '==> Windows Firewall rules - Press return to continue'
-        $input = Read-Host -Prompt 'Press return to continue'
+        Write-Output '==> Windows Firewall rules'
+        if (!$NoPrompt) { $input = Read-Host -Prompt 'Press return to continue' }
 
         $hercules_exedir = Resolve-Path "$hercules_dir\$Flavor\msvc.AMD64.bin"
         # Write-Output "Hercules exe directory : $hercules_exedir"
@@ -893,7 +896,7 @@ try {
 
                     Write-Output ""
                     Write-Output "Windows Firewall rules have been created."
-                    $input = Read-Host -Prompt 'Press return to continue'
+                    if (!$NoPrompt) { $input = Read-Host -Prompt 'Press return to continue' }
                 }
             }
         }
diff --git a/hercules-step2.cmd b/hercules-step2.cmd
index d41414b..9be9acd 100644
--- a/hercules-step2.cmd
+++ b/hercules-step2.cmd
@@ -15,8 +15,8 @@
 :: Called from hercules-buildall.ps1, from the hercules-helper\windows directory
 
 @if defined TRACEON (@echo on) else (@echo off)
-
-echo Starting Hercules-Step 2...
+set NOPROMPT=%~2
+@echo Starting Hercules-Step 2...
 
 pushd %HERCULES_HELPER_BUILD_DIR%\%1
 
@@ -29,17 +29,21 @@ pushd %HERCULES_HELPER_BUILD_DIR%\%1
     :: set INCLUDE=%INCLUDE%;C:\Program Files (x86)\Microsoft SDKs\Windows\v7.1A\Include
 
     echo.
-    echo [40;92m==^> Build Hercules - Press return to continue ...[0m
+    echo [40;92m==^> Build Hercules[0m
+    if "%NOPROMPT%" neq "True" (
+        echo [40;92m==^Press return to continue ...[0m
         set /P dummy=
-
+    )
     call makefile.bat RETAIL-X64 makefile.msvc 8 -title "*** Hercules-Helper Test Build ***" -a
 
     echo.
     echo [40;92m Just FYI: Windows Defender anti-malware may cause the tests to fail or hang. [0m
     echo.
-    echo [40;92m==^> Run Tests - Press return to continue ...[0m
+    echo [40;92m==^> Run Tests [0m
+    if "%NOPROMPT%" neq "True" (
+        echo [40;92m==^Press return to continue ...[0m
         set /P dummy=
-
+    )
     call tests\runtest.cmd -n * -t 2 -d ..\%1\tests
 
 popd
```

wrljet commented:
I've been a little busy but I'll get this tested locally soon.
Thank you for the submissions!
Bill

wrljet commented:
Ross,

I have finally added your changes.  I had made other mods, and it goofed up the diff, so I incorporated your code by hand.
I also removed all my tab characters as a separate change first.

This is current with Commit https://github.com/wrljet/hercules-helper-windows/commit/5b1f740700e1988a5797451193d7a0d14366cfc1

Thank you for the feature idea and the code.

Bill
